frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/factorial-experiment-design.component.ts

```diff
@@ -0,0 +1,603 @@
+
+    if (!this.conditionTableDataUpToDate) {
+      this.experimentDesignStepperService.updateFactorialDesignData(this.factorialExperimentDesignForm.value);
+    }
+
+    if (this.isFormValid()) {
+      const factorialExperimentDesignFormData = this.factorialExperimentDesignForm.value;
+      const factorialPartitions = this.convertToPartitionData(factorialExperimentDesignFormData);
+      const factorialConditions = this.experimentDesignStepperService.createFactorialConditionRequestObject();
+
+      const factorialConditionAliases: ExperimentConditionAliasRequestObject[] =
+        this.experimentDesignStepperService.createFactorialConditionsConditionAliasesRequestObject();
+
+      this.emitExperimentDialogEvent.emit({
+        type: eventType,
+        formData: {
+          conditions: factorialConditions,
+          partitions: factorialPartitions,
+          conditionAliases: factorialConditionAliases,
+        },
+        path: NewExperimentPaths.EXPERIMENT_DESIGN,
+      });
+
+      // scroll back to the factors table
+      this.scrollToFactorsTable();
+    }
+  }
+
+  convertToPartitionData(factorialExperimentDesignFormData) {
+    let order = 1;
+    let factorOrder = 1;
+    const Partitions = [];
+    this.levelIds = [];
+    factorialExperimentDesignFormData.factors.forEach((partition) => {
+      let levelOrder = 1;
+      const currentLevels: ExperimentLevel[] = partition.levels.map((level) => {
+        return { name: level.level, alias: level.alias, id: level.id, order: levelOrder++ };
+      });
+      const currentFactors: ExperimentFactor = {
+        name: partition.factor,
+        order: factorOrder++,
+        levels: currentLevels,
+      };
+      if (
+        !Partitions.find(
+          (existingPartition) =>
+            existingPartition.site === partition.site && existingPartition.target === partition.target
+        )?.factors.push(currentFactors)
+      ) {
+        const partitionData = {
+          site: partition.site,
+          id: uuidv4(),
+          description: '',
+          order: order++,
+          excludeIfReached: false,
+          factors: [currentFactors],
+        };
+        partition.target
+          ? Partitions.push({ ...partitionData, target: partition.target })
+          : Partitions.push(partitionData);
+      }
+    });
+    return Partitions;
+  }
+
+  scrollToFactorsTable(): void {
+    this.stepContainer.nativeElement.scroll({
+      top: 0,
+      behavior: 'smooth',
+      duration: 500,
+      easing: 'easeOutCubic',
+    });
+  }
+
+  scrollToConditionsTable(): void {
+    this.conditionTableDataUpToDate = true;
+    this.stepContainer.nativeElement.scroll({
+      top: this.stepContainer.nativeElement.scrollHeight / 2,
+      behavior: 'smooth',
+      duration: 500,
+      easing: 'easeOutCubic',
+    });
+  }
+
+  get factor(): FormArray {
+    return this.factorialExperimentDesignForm?.get('factors') as FormArray;
+  }
+
+  get level(): FormArray {
```

Comment:
this should be dead code and can be removed

Reply:
no level are called and it will give error if removed

Reply:
ok i see, I was thinking we had replaced this with `getLevels()`, but it's different.